### ISSUE TITLE
The fusion module is discarded

### ISSUE THREAD

**Issue by qinliuliuqin** [hkchengrex/XMem#17]
Hi Rex, 
Congrats for your another great work!!
I have been working on applying STCN for interactive segmentation on 3D medical images ([MICCAI'22 iSegFormer](https://github.com/uncbiag/iSegFormer)), and now it seems XMem is a better choice considering its memory efficiency.
However, since you discarded fusion model in XMem, wouldn't it cause performance to drop because the user's intention will be ignored when he/she refines the mask and starts a new propagation? Do you think it's necessary to retrain the fusion model? 
Thank you in advance!

Best,
Qin

**Comment by hkchengrex** [comment]
Hi, thank you for your interest. I have seen your work before and I think that's a great use of iVOS which I haven't thought of before. 

We did not pay a lot of attention to the interactive setting in XMem. I simply do not have the time/energy to train the fusion module and write the corresponding code for XMem. You're right that having a fusion module would probably make it better. Whether it is *necessary* would depend on your use case. I would suggest trying without first as it is always easier :)

**Comment by qinliuliuqin** [comment]
Thanks for your comments. 
Actually, I tried XMem on medical images once you released the code. My observation is that XMem works better than STCN for the first propagation, but does not work as good as STCN in the interactive setting. That's why I guess the fusion module might be necessary for the interactive setting.

Both STCN and XMem work surprisingly well on large and regular-shaped organs (e.g., cardiac, lung, liver, and kidney), without finetuning, but they share similar failure patterns on small and tubular-structured organs (e.g., rib and vessels). Besides, these failure cases are difficult to refine in the interactive setting (I may post some videos showing these cases in this thread later). I have improved STCN a little bit on medical images with some known tricks (e.g., finetuning on medical images and [cycle-consistency loss](https://proceedings.neurips.cc//paper/2020/file/0d5bd023a3ee11c7abca5b42a93c4866-Paper.pdf)). However, as you mentioned [here](https://github.com/hkchengrex/XMem/blob/main/docs/FAILURE_CASES.md), more positional and motion clues are required to better handle these failure cases.
